### PR TITLE
Refactored the Code and improved the Code Quality and Performance

### PR DESCRIPTION
**Pull request by HarshCasper**
This pull request fixes some of the quality issues raised by DeepSource on my fork of this repository.

Summary of fixes:

- Change methods not using its bound instance to staticmethods
- Refactor unnecessary `else` / `elif` when `if` block has a `return` statement
- Change methods not using its bound instance to staticmethods 
- Remove unused private method
- Useless inheritance from object
- Remove implicit `object` from the base class

You can see all the issues raised by DeepSource [here](https://deepsource.io/gh/HarshCasper/sugar).

These Issues were fixed automatically by DeepSource

### Using DeepSource to continuously analyze your repository

- Merge this PR. I have included a `.deepsource.toml` in this PR, which you can use to configure your analysis settings.
- Install DeepSource on your repository [here](https://deepsource.io/signup).
- Activate analysis [here](https://deepsource.io/gh/sugarlabs/sugar).

If you do not want to use DeepSource to continuously analyze this repo, I'll remove the `.deepsource.toml` from this PR and you can merge the rest of the fixes.

### CODE REVIEW

**Comment by srevinsaju** [comment]
Hello @HarshCasper 

Thanks for your PR

I have a few questions
* Are these changes made by you, i.e @HarshCasper ? If yes, could you split the changes into smaller commits with the fixes as commit messages so that its easier to understand?
* Are these changes made by a bot / SaaS? if yes, I guess we should attribute this commit to the bot @deepsourcebot or @deepsource-autofix

**Comment by HarshCasper** [comment]
Hi @srevinsaju 

Thanks for taking the time to review this. The Commits were made by the DeepSource App and I pushed all the changes to a Single Commit. 

If you would like, I can split the changes into smaller commits and send individual PRs for them to make it easy for them to review. If there is not a hard-fast requirement to squash the Commits before the merge, I will have the commits attributed to the Bot. 

Since the DeepSource Analysis has issues that feature issues that can be auto fixed or can be just manually fixed, I would love to work on the same.

File: extensions/cpsection/aboutme/view.py [hunk 1]
```diff
@@ -418,5 +418,6 @@ def __gender_changed_cb(self, genderpicker, gender):
         self._age_pickers.update_gender(gender)
         return False
 
-    def __age_changed_cb(self, event, age):
+    @staticmethod
```
Comment (chimosky) [inline extensions/cpsection/aboutme/view.py:421]: Why make this method a static method?

**Review by chimosky — COMMENTED** [review]
Haven't reviewed all the changes made, diff is quite large.

File: extensions/cpsection/aboutme/view.py (hunk 1 above)
Comment (srevinsaju) [reply extensions/cpsection/aboutme/view.py:421]: parameter `self` is passed, but not used.

**Comment by HarshCasper** [comment]
@chimosky I guess it would be best, if I close this PR and start work on a fresh one adding discrete changes to the same. Kindly let me know if that would work.

**Comment by chimosky** [comment]
> @chimosky I guess it would be best, if I close this PR and start work on a fresh one adding discrete changes to the same. Kindly let me know if that would work.

That would be great, thanks.

**Comment by quozl** [comment]
Thanks.  Reviewed.

* https://github.com/sugarlabs/sugar/commit/d59b62a5f1db8c1e1ecb7ccbbcc8b85163290908 was based off d59b62a5f1db8c1e1ecb7ccbbcc8b85163290908 instead of the current master at the time b10835f5c1f5388c82f7570a0af94d11010482e8 so there were changes shown in the diff that were not what you intended to present; please rebase next time,
* the changes make maintenance of the Python 2 branch more complicated for me, but I think I can cope,
* the changes do not have substantial quality or performance value, but we do accept _flake8_ and _pep8_ changes without worrying about this; it is unfortunate that we spend our time on such automaton changes rather than improve Sugar with new features, of which we have plenty of opportunities and nobody working on them,
* the addition of `@staticmethod` makes the source longer without adding substantive value, although the removal of `else` has compensated somewhat,
* the removal of `else` in places that use a premature `return` is something I like, but we've had a history of sticking to classical structured coding, in the hope that our users may discover how Sugar works (hence the View Source feature and the Pippy activity),
* some indentation changes have crept in, e.g. telepathyclient.py,
* merging this would make Sugar style quite different to the Toolkit and activities source,
* no, I'm not interested in ongoing analysis, please work on something real and useful.

I've cherry-picked 7bbff9f5981("Deepsource: unnecessary `else` or `elif`") and 0388f516b ("Deepsource: Remove unused private method") and pushed to master.

**Pull request closed without merging**